docs/guides/table_migration.md

```diff
@@ -0,0 +1,120 @@
+# Table migration
+
+SQLMesh projects can read directly from tables not managed by SQLMesh, but in some cases it may be desirable to migrate an existing table into a SQLMesh project. This guide describes two methods for migrating existing tables into a SQLMesh project.
+
+We assume that downstream consumers will continue to select from an object with the original table name, so the migration task is to convert an existing table into an identically named SQLMesh model. We also assume that the table is incremental, as there would be no need for migration if the table could be rebuilt on every run.
+
+## Do you need to migrate?
+
+SQLMesh does not assume it manages all data sources: models can read from any data source accessible by the SQL engine, treating them as as [external models](../concepts/models/model_kinds.md#external) that include column-level lineage or as generic sources. This approach is preferred to migrating existing tables into a SQLMesh project.
+
+You should only migrate tables if the source table you are working with is either too large to be rebuilt or cannot be rebuilt because the necessary historical data is unavailable.
+
+We describe two migration methods below. The stage and union migration method is preferred and should be used if it is feasible to store both the original table and all its rows copied into another table. If the table is too large to duplicate, the snapshot replacement method can be used.
+
+## Stage and union
+
+The stage and union method works by treating historical and new data as separate sources. It requires creating a staging model to ingest new records and a primary model that unions those records with the existing table's historical records.
+
+For example, consider an existing table named `my_schema.existing_table`. Migrating this table with the stage and union method consists of five steps:
+
+1. Ensure `my_schema.existing_table` is up to date (has ingested all available source data)
+2. Rename `my_schema.existing_table` to any other name, such as `my_schema.existing_table_historical`
+3. Create a new staging [`INCREMENTAL_BY_TIME_RANGE` model](../concepts/models/model_kinds.md#incremental_by_time_range) named `my_schema.existing_table_staging` (see below for code).
+4. Create a new primary [`INCREMENTAL_BY_TIME_RANGE` model](../concepts/models/model_kinds.md#incremental_by_time_range) named `my_schema.existing_table` (see below for code).
```

Comment:
Should this be a `VIEW` model instead? Otherwise we don't really save compute and there isn't a point in having a staging model IMHO. In my mind staging model is needed for new data only, while this becomes a view over both historical and new data.